fintrack/transactions/tasks.py

```diff
@@ -0,0 +1,40 @@
+from celery import shared_task
+from decouple import config
+from django.core import mail
+from django.utils import timezone
+
+from .models import Transaction
+
+@shared_task
+def send_transaction_emails():         
+    transactions = Transaction.objects.filter(
+        date=timezone.now().date()
+    )
+    user_emails = transactions.values_list(
+        'transaction_from_account__holder__email',
+        flat=True
+    ).distinct()
+
+    messages = []
+
+    for user_email in user_emails:
+        user_transactions = transactions.filter(
+            transaction_from_account__holder__email=user_email
```

Comment:
it's always better to user foreign key to fetch related objects or related name to do that